Projects/IntelligentObjectAvoider/robot.py

```diff
@@ -0,0 +1,352 @@
+            put_on_hold.clear()
+
+        # if we need to move the GoPiGo3 around
+        if previous != 0 and distance_to_walk >= 0:
+            gopigo3_robot.drive_cm(distance_to_walk * how_much_of_distance)
+
+        # keep the previous distance we read
+        # if it's 0 now and the next time is still 0
+        # then the drive method won't be called
+        previous = distance_to_walk
+
+        sleep(0.001)
+
+    # stop the motors when the thread is being stopped
+    gopigo3_robot.stop()
+
+
+def Main(trigger):
+    print("   _____       _____ _  _____         ____  ")
+    print("  / ____|     |  __ (_)/ ____|       |___ \ ")
+    print(" | |  __  ___ | |__) || |  __  ___     __) |")
+    print(" | | |_ |/ _ \|  ___/ | | |_ |/ _ \   |__ < ")
+    print(" | |__| | (_) | |   | | |__| | (_) |  ___) |")
+    print("  \_____|\___/|_|   |_|\_____|\___/  |____/ ")
+    print("                                            ")
+
+    print("Let your GoPiGo3 move around and avoid any obstacles.")
+    print("Pay attention to how your GoPiGo3 moves around.")
+    print("Avoid sharp corners / edges as the algorithm wasn't made for advanced stuff.")
+
+    # Event object for letting one thread
+    # control the other thread's flow control
+    put_on_hold = threading.Event()
+    # used for synchronizing the threads
+    simultaneous_launcher = threading.Barrier(2)
+    # for exchanging messages between threads
+    sensor_queue = queue.Queue()
+
+    print("\nWaiting threads to fire up")
+
+    path_finder = threading.Thread(target=obstacleFinder, args=(trigger, put_on_hold, simultaneous_launcher, sensor_queue))
+    controller = threading.Thread(target=robotController, args=(trigger, put_on_hold, simultaneous_launcher, sensor_queue))
+
+    # start the threads
+    path_finder.start()
+    controller.start()
+
+    # wait for the user to press CTRL-C
+    # or to have an error while firing up the threads
+    while not trigger.is_set() and not simultaneous_launcher.broken:
+        sleep(0.001)
+    # if an error was encountered
+    if simultaneous_launcher.broken:
+        # then exit the script
+        sys.exit(1)
+
+    # otherwise, just wait for the threads to finish
+    path_finder.join()
+    controller.join()
+
+    # and then exit successfully
+    sys.exit(0)
+
+if __name__ == "__main__":
+    exit_trigger = threading.Event()
+    # CTRL-C signal handler
+    # used a lambda, because a proper function is too much code
+    signal.signal(signal.SIGINT, lambda signum, frame: exit_trigger.set())
+    Main(exit_trigger)
+
+##################################
+# Heavy-math project on avoiding #
+# objects while continuously     #
+# reading with the distance      #
+# sensor - vezi desen scanat     #
+##################################
+"""
+def offset_movement(values):
+
+    # distanta maxima de intoarcere permisa pe o intoarcere completa a servomoturlui este limitata la 40 de grade.
```

Comment:
First of all, these comments are written in a commented section, which doesn't interfere with the actual code.

The second thing is, these comments are just for me. 
Initially, I was working on a solution which proved to be quite complex and because I was time-restrained I had to choose a faster alternative. These comments describe some trigonometry-related math stuff. All the math I've done in college and high school was in Romanian and translating it to English would be really difficult and not natural, since I don't have that lexicon.

Since I'm busy with other stuff at the moment, I decided to put this on hold until I come back and start working on it again.